File: coding-for-fun/elixir/fahrenheit_to_celsius.exs
# author: https://github.com/AnotherGithubDude, 2022
# licence: cc0
# tested with Elixir 1.14.2 (compiled with Erlang/OTP 25)

# Prompt user to input the Fahrenheit value from command-line
# Formula example: (74°F − 32) × 5/9 = 23,333°C
f_input = String.trim(IO.gets("Please enter the temperature in Fahrenheit (84.0, 68.2):\t"))

# Convert the Fahrenheit input to Celsius and save the result
pre_result = String.to_float(f_input)
result = (pre_result - 32) * (5/9)

# Print the result as output on the terminal 
IO.puts("The temperature in Celsius is: #{result}")

File: coding-for-fun/rust/fahrenheit_to_celsius.rs
// author: https://github.com/AnotherGithubDude, 2022
// tested with rustc 1.6.5

use std::io;

fn main() {

    println!("Enter Fahrenheit Value:");

    let mut fahrenheit_input = String::new();

    io::stdin().read_line(&mut fahrenheit_input)
        .expect("Error reading line");

    let fahrenheit_input: f64 = fahrenheit_input.trim().parse()
        .expect("Error parsing input");

    let celsius_value = (fahrenheit_input - 32.0) * (5.0/9.0);

    println!("The Celsius value is {}", celsius_value);
}
